### PR TITLE
Ret 2205

### PR DESCRIPTION
**Pull request by hsjhita**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-2205

### Change description ###

Added doc gen for ET3 Vetting/Processing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java [hunk 1]
```diff
@@ -961,4 +961,25 @@ public static Address getRespondentAddressET3(RespondentSumType respondentSumTyp
 
     }
 
+    /**
+     * Utility method to null check a string value.
+     * @param value to be checked if null or contains data
+     * @return null if the value is null or empty or will return the value itself
+     */
+    public static String nullChecker(String value) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java:969]: 1. Can't we reuse something instead? Is there no library that we can use to do this? What about StringUtils.defaultIfEmpty
2. I don't think DocumentHelper is an appropriate place for this method or listNullChecker. They are generic utility methods with nothing that ties them specifically to documents. They belong in a different class.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3VettingService.java [hunk 2]
```diff
@@ -90,4 +103,13 @@ private Et3VettingType getEt3VettingTypeForRespondent(RespondentSumTypeItem resp
     private Et3VettingType copyEt3FieldsFromCaseDataToRespondent(CaseData caseData) {
         return (Et3VettingType) intersectProperties(caseData, Et3VettingType.class);
     }
+
+    public DocumentInfo generateEt3ProcessingDocument(CaseData caseData, String userToken, String caseTypeId) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3VettingService.java:107]: Javadoc

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java [hunk 3]
```diff
@@ -256,11 +258,22 @@ public DocumentInfo generateEt1VettingDocument(CaseData caseData, String userTok
...
+                case "ET1 Vetting.pdf" :
+                    outputStreamWriter.write(Et1VettingHelper.getDocumentRequest(caseData,
+                            tornadoConnection.getAccessKey()));
+                    break;
+                case "ET3 Processing.pdf" :
+                    outputStreamWriter.write(Et3VettingHelper.getDocumentRequest(caseData,
+                            tornadoConnection.getAccessKey()));
+                    break;
+                default:
+                    throw new NotFoundException("Document not found");
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java:275]: This is the wrong exception to throw. It's a webjars exception and nothing to do with what the code is doing. How about
`throw new IllegalArgumentException("Unexpected document name " + documentName);`

Comment (jthmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java:275]: Also you should check the documentName argument at the start of the method

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java (hunk 1 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java]: Replaced with defaultIsEmpty, generic utility methods removed from code

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3VettingService.java (hunk 2 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3VettingService.java:114]: Javadoc added

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java (hunk 3 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java]: Corrected exception and added one at the start of the method - clarification on the duplication if possible?

Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java]: I think I know what you mean. Both methods are creating the full request string but you think there's duplication over some properties? Out of the 4 top level fields, only 1 is the same - the access key, the rest change as it's specific to each document. I think that's what you mean, if not you'll need to clarify

Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java]: Duplication removed

**Review by jthmcts — APPROVED** [review]

**Review by robin-ieong — APPROVED** [review]

**Pull request merged**